x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_editing/index.tsx

```diff
@@ -172,7 +175,20 @@ const EditRulePageComponent: FC<{ rule: RuleResponse }> = ({ rule }) => {
     newTermsFields: defineStepData.newTermsFields,
   });
 
-  const loading = userInfoLoading || listsConfigLoading;
+  const { upgradeReviewResponse, isLoading: isRuleUpgradeReviewLoading } = usePrebuiltRulesUpgrade({
+    pagination: {
+      page: 1, // we only want to fetch one result
+      perPage: 1,
+    },
+    filter: { rule_ids: [ruleId] },
```

Comment:
request `/internal/detection_engine/prebuilt_rules/upgrade/_review` is getting send even for custom rules.

Can we avoid this? We already have `rule.immutable` parameter in this component. Can it be used to avoid unnecessary requests?

Same goes for rule details page and any other that can be optimized